### ISSUE TITLE
solc 0.4.10 compiles code but not abi

### ISSUE THREAD

**Issue by islandBilly** [ethereum/solidity#1826]
I downloaded the solc binary from your release page and moved into /usr/local/bin. Then I started my geth node and told it where solc was, which worked great! It even remembers from session to session. But when I try to compile a contract, either in a web3 js page or in the geth js console, I get an empty abi object:

> var contractCompiled = eth.compile.solidity(contract);
undefined
> contractCompiled
{
  /tmp/geth-compile-solidity988608421:BallotItem0_1_0: {
    code: "0x6060604052341561000c57fe5b5b60008054600160a060020a03191673e4ee86b81d35838adde358d66c8b2646e6c9270917905565015c06d45e0060015565015c5bcd4c0060025560408051808201909152600f8082527f5768617420746865204675636b3f3f0000000000000000000000000000000000602090920191825261008b9160039161009c565b50600160045560006005555b61013c565b828054600181600116156101000203166002900490600052602060002090601f016020900481019282601f106100dd57805160ff191683800117855561010a565b8280016001018555821561010a579182015b8281111561010a5782518255916020019190600101906100ef565b5b5061011792915061011b565b5090565b61013991905b808211156101175760008155600101610121565b5090565b90565b6105268061014b6000396000f300606060405236156100725763ffffffff60e060020a60003504166334d6087d81146100745780635fa520bb146100965780636a6396f2146100ee5780637284e416146101105780638d486837146101a05780639a3fc7ec146101bb578063cebdaea6146101dd578063f6fd7fde146101ff575bfe5b341561007c57fe5b61008461029c565b60408051918252519081900360200190f35b341561009e57fe5b6100ec600480803590602001908201803590602001908080601f016020809104026020016040519081016040528093929190818152602001838380828437509496506102a295505050505050565b005b34156100f657fe5b6100846102fa565b60408051918252519081900360200190f35b341561011857fe5b610120610300565b604080516020808252835181830152835191928392908301918501908083838215610166575b80518252602083111561016657601f199092019160209182019101610146565b505050905090810190601f1680156101925780820380516001836020036101000a031916815260200191505b509250505060405180910390f35b34156101a857fe5b6100ec60043560243560443561038e565b005b34156101c357fe5b610084610413565b60408051918252519081900360200190f35b34156101e557fe5b610084610419565b60408051918252519081900360200190f35b341561020757fe5b61021260043561041f565b60408051602080825283546002600019610100600184161502019091160490820181905290918291908201908490801561028d5780601f106102625761010080835404028352916020019161028d565b820191906000526020600020905b81548152906001019060200180831161027057829003601f168201915b50509250505060405180910390f35b60045481565b6000543373ffffffffffffffffffffffffffffffffffffffff9081169116146102cb5760006000fd5b6005805460018101909155600090815260066020908152604090912082516102f592840190610430565b505b50565b60025481565b6003805460408051602060026001851615610100026000190190941693909304601f810184900484028201840190925281815292918301828280156103865780601f1061035b57610100808354040283529160200191610386565b820191906000526020600020905b81548152906001019060200180831161036957829003601f168201915b505050505081565b6000543373ffffffffffffffffffffffffffffffffffffffff9081169116146103b75760006000fd5b8160015411156103c75760006000fd5b8160025410156103d75760006000fd5b600081815260066020526040902060019081018054909181016103fa83826104af565b916000526020600020900160005b50849055505b505050565b60015481565b60055481565b600660205260009081526040902081565b828054600181600116156101000203166002900490600052602060002090601f016020900481019282601f1061047157805160ff191683800117855561049e565b8280016001018555821561049e579182015b8281111561049e578251825591602001919060010190610483565b5b506104ab9291506104d9565b5090565b81548183558181151161040e5760008381526020902061040e9181019083016104d9565b5b505050565b6104f791905b808211156104ab57600081556001016104df565b5090565b905600a165627a7a72305820badc4022dd5d9219e7690336f99f407a866b3018c05c94fced0eb5a2665230e50029",
    info: {
      abiDefinition: [{...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}, {...}],
      compilerOptions: "--combined-json bin,abi,userdoc,devdoc --add-std --optimize",
      compilerVersion: "0.4.10",
      developerDoc: {
        methods: {}
      },
      language: "Solidity",
      languageVersion: "0.4.10", etc (this is from the geth js console)

I tried using the --abi parameter on a compile, but it is flagged as an error. I need to programmatically compile small contracts in a web3 and then go on to run some transactions against them.

Any idea why this is happening?

**Comment by chriseth** [comment]
@islandBilly thanks for reporting! You should perhaps create a similar issues in the go-ethereum repository, as this might as well be about how solc is used from geth.
On the other hand, we do not recommend to use `web3.eth.compile` anymore, because the interface is complex and inflexible at the same time. If you already are in a javascript environment, please consider using solcjs: http://github.com/ethereum/solcjs

**Comment by islandBilly** [comment]
Okay, Chris, thanks. I have put my issue over on the go-ethereum repository as you suggested, but this sure looks to me like a compiler problem. Geth has decided to invoke it with certain options, but when source has no errors, how can the compiler put out bytecodes but no abi codes? I tried it with the greeter contract code with the same results.
   I have also installed npm and used it to install solc-js, but now I can't seem to do a successful require('solc') in my code. Will go over to the solc-js folks to ask about that.
